java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -3509,9 +3509,9 @@ void testCastFloatToDecimal() {
   @Test
   void testCastDoubleToDecimal() {
     testCastNumericToDecimalsAndBack(DType.FLOAT64, false, 0,
-        () -> ColumnVector.fromBoxedDoubles(1.0, 2.1, -3.23, null, 2.41281, (double) Long.MAX_VALUE),
-        () -> ColumnVector.fromBoxedDoubles(1.0, 2.0, -3.0, null, 2.0, (double) Long.MAX_VALUE),
-        new Long[]{1L, 2L, -3L, null, 2L, Long.MAX_VALUE}
+        () -> ColumnVector.fromBoxedDoubles(1.0, 2.1, -3.23, null, 2.41281, (double) Integer.MAX_VALUE),
+        () -> ColumnVector.fromBoxedDoubles(1.0, 2.0, -3.0, null, 2.0, (double) Integer.MAX_VALUE),
+        new Long[]{1L, 2L, -3L, null, 2L, Integer.MAX_VALUE}
```

Comment:
```suggestion
        new Long[]{1L, 2L, -3L, null, 2L, Long.MAX_VALUE}
```

Reply:
Long.MAX_VALUE is the max dec64 value, but an input of (double)Long.Max_VALUE is 2^63, greater than what can be stored in dec64, so a test of it is not valid.  See [this comment](https://github.com/rapidsai/cudf/pull/15905#discussion_r1657903728) for discussion.

Reply:
Note that it is a valid test -- there's no such thing as a "bad" value in a test, just what is expected when that value is processed.  That's why I asked _why_ this was changed.  The test used to pass, and Java does the "impossible" here: 
```
scala> Long.MaxValue
res2: Long = 9223372036854775807

scala> Long.MaxValue.toDouble
res3: Double = 9.223372036854776E18

scala> Long.MaxValue.toDouble.toLong
res4: Long = 9223372036854775807
```
The old behavior matched the Java behavior, so technically this is a regression in behavior, at least from the spark-rapids perspective.  Given how much better other cases are handled, it seems like this is a net benefit, but this example shows it's not always producing better values than the old approach.